EyeSeeTea/dhis2-light-sdk
# Feature add api version parameter

### :pushpin: References
* **Issue:** close https://github.com/EyeSeeTea/malariapp/issues/2212
* **Related pull-requests:** https://github.com/EyeSeeTea/malariapp/pull/2214

### :tophat: What is the goal?

Create request to endoint /system/info and creates apiVersion parameter for D2Api

###   :gear: branches 
**dhis2-light-sdk**:
       Origin: feature-add_api_version_parameter Target: developent

### :memo: How is it being implemented?

* I have added a SystemInfo endpoint class, models and tests for clients can retrieve api version
* Second I have added apiVersion to include if is not empty in requests

### :boom: How can it be tested?

 **Use case 1:** execute code analysis and execute all tests with this command:
```
./gradlew build
```

**Use case 2:** execute Android sample app, insert url, ApiVersion, username, password and click on the button to realize request against dhis2 server. The response toString method should be rendered on the screen

**Use case 3:** execute Web sample

1. Open chrome disabling web security:

For OSX:
```
$ open -a Google\ Chrome --args --disable-web-security --user-data-dir
```
--user-data-dir required on Chrome 49+ on OSX

For Linux run:
```
$ google-chrome --disable-web-security
```

For Windows go into the command prompt and go into the folder where Chrome.exe is and type:
```
chrome.exe --disable-web-security
```

2. Compile code to js and create bundle.js

execute:
```
./gradlew buildJsToWeb
```
3. Open web sample
Open index.html under webSample directory, insert URL, ApiVersion, username, password and click on the button to realize request against dhis2 server. The response toString method should be rendered on the screen

### :floppy_disk: Requires DB migration?


### :art: UI changes?


<img width="600" alt="websample" src="https://user-images.githubusercontent.com/5593590/50397635-30542d80-0772-11e9-80c6-4f579d17a24d.png">

<img width="300" alt="websample" src="https://user-images.githubusercontent.com/5593590/50397650-48c44800-0772-11e9-8f32-8415209a1847.png">
